testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py

```diff
@@ -3787,6 +3789,124 @@ def test_replication_behavior_with_channelRole_modification(params_from_base_tes
+    'byte_array',
+    'stream',
+    'file_url'
+])
+def test_blob_contructor_replication(params_from_base_test_setup, blob_data_type):
+    '''
+    @summary:
+    1. Create docs in CBL
+    2. Do push replication
+    3. update docs in CBL with attachment in specified blob type
+    4. Do push replication after docs update
+    5. Verify blob content replicated successfully
+    '''
+    cbl_db = params_from_base_test_setup["source_db"]
+    db = params_from_base_test_setup["db"]
+    base_url = params_from_base_test_setup["base_url"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    liteserv_platform = params_from_base_test_setup["liteserv_platform"]
+
+    # Reset cluster to ensure no data in system
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    sg_db = "db"
+    num_of_docs = 10
+    channels = ["ABC"]
+    username = "autotest"
+    password = "password"
+
+    sg_client = MobileRestClient()
+    sg_client.create_user(sg_admin_url, sg_db, username, password=password, channels=channels)
+    cookie, session_id = sg_client.create_session(sg_admin_url, sg_db, username)
+    session = cookie, session_id
+
+    # 1. Create docs in CBL
+    db.create_bulk_docs(num_of_docs, "cbl_sync", db=cbl_db, channels=channels)
+
+    # 2. Do push replication
+    replicator = Replication(base_url)
+    authenticator = Authenticator(base_url)
+    replicator_authenticator = authenticator.authentication(session_id, cookie, authentication_type="session")
+    repl = replicator.configure_and_replicate(source_db=cbl_db,
+                                              target_url=sg_blip_url,
+                                              continuous=True,
+                                              replicator_authenticator=replicator_authenticator,
+                                              replication_type="push")
+    replicator.stop(repl)
+
+    sg_docs = sg_client.get_all_docs(url=sg_admin_url, db=sg_db, include_docs=True)["rows"]
+    # Verify database doc counts
+    cbl_doc_count = db.getCount(cbl_db)
+    assert len(sg_docs) == cbl_doc_count, "Expected number of docs does not exist in sync-gateway after replication"
+
+    # 3. update docs in CBL with attachment in specified blob type
+    blob = Blob(base_url)
+    dictionary = Dictionary(base_url)
+
+    doc_ids = db.getDocIds(cbl_db)
+    cbl_db_docs = db.getDocuments(cbl_db, doc_ids)
+    for doc_id, doc_body in list(cbl_db_docs.items()):
+        mutable_dictionary = dictionary.toMutableDictionary(doc_body)
+        dictionary.setString(mutable_dictionary, "new_field_string_1", random_string(length=30))
+        dictionary.setString(mutable_dictionary, "new_field_string_2", random_string(length=80))
+
+        if liteserv_platform == "android":
```

Comment:
From line 3862 to 3888, we have to replace with utils or common method  as we might use this code repeatedly later  when we add more blob tests. you can leave it now. We can add as action item and make the change later

Reply:
added a new method in utils, make this code for generic use